FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc

```diff
@@ -166,89 +129,17 @@ TrajectorySeedProducer::beginRun(edm::Run const&, const edm::EventSetup & es)
 }
 
 bool
-TrajectorySeedProducer::passSimTrackQualityCuts(const SimTrack& theSimTrack, const SimVertex& theSimVertex) const
+  TrajectorySeedProducer::pass2HitsCuts(const TrajectorySeedHitCandidate& hit1, const TrajectorySeedHitCandidate& hit2) const
 {
-    //require min pT of the simtrack
-    if ((simTrack_pTMin>0) && ( theSimTrack.momentum().Perp2() < simTrack_pTMin*simTrack_pTMin))
-    {
-        return false;
-    }
-    if(((simTrack_maxD0<0) && (simTrack_maxZ0<0)) || (!testPrimaryVertexCompatibility && !testBeamspotCompatibility))
-      {
-	return true;
-      }
-  
-    //require impact parameter of the simtrack
-    BaseParticlePropagator theParticle = BaseParticlePropagator(
-        RawParticle(
-            XYZTLorentzVector(
-                theSimTrack.momentum().px(),
-                theSimTrack.momentum().py(),
-                theSimTrack.momentum().pz(),
-                theSimTrack.momentum().e()
-            ),
-            XYZTLorentzVector(
-                theSimVertex.position().x(),
-                theSimVertex.position().y(),
-                theSimVertex.position().z(),
-                theSimVertex.position().t())
-            ),
-            0.,0.,4.
-    );
-    theParticle.setCharge(theSimTrack.charge());
-
-
-    //this are just some cuts on the SimTrack for speed up
-    std::vector<const math::XYZPoint*> origins;
-    if (testBeamspotCompatibility)
-    {
-        origins.push_back(&beamSpot->position());
-    }
-    if (testPrimaryVertexCompatibility)
-    {
-        for (unsigned int iv = 0; iv < primaryVertices->size(); ++iv)
-        {
-            origins.push_back(&(*primaryVertices)[iv].position());
-        }
-    }
-
-    //only one possible origin is required to succeed
-    for (unsigned int i = 0; i < origins.size(); ++i)
+  if(theRegionProducer){
+    return testWithRegions(hit1,hit2);
+  }
+  else 
     {
-        if ((simTrack_maxD0>0.0) && ( theParticle.xyImpactParameter(origins[i]->X(),origins[i]->Y()) > simTrack_maxD0 ))
-        {
-            continue;
-        }
-        if ((simTrack_maxZ0>0.0) && ( fabs( theParticle.zImpactParameter(origins[i]->X(),origins[i]->Y()) - origins[i]->Z()) > simTrack_maxZ0))
-        {
-            continue;
-        }
-        return true;
+      std::cout<<"Either region producer, or the seed creator cfg is not available."<<std::endl;
```

Comment:
Sure !!☺

On Fri, Aug 14, 2015 at 5:38 PM, lveldere <notifications@github.com> wrote:

> In FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc
> <https://github.com/cms-sw/cmssw/pull/10763#discussion_r37070297>:
>
> >      {
> > -        if ((simTrack_maxD0>0.0) && ( theParticle.xyImpactParameter(origins[i]->X(),origins[i]->Y()) > simTrack_maxD0 ))
> > -        {
> > -            continue;
> > -        }
> > -        if ((simTrack_maxZ0>0.0) && ( fabs( theParticle.zImpactParameter(origins[i]->X(),origins[i]->Y()) - origins[i]->Z()) > simTrack_maxZ0))
> > -        {
> > -            continue;
> > -        }
> > -        return true;
> > +      std::cout<<"Either region producer, or the seed creator cfg is not available."<<std::endl;
>
> Hi Anshul
>
> You are right, and I corrected a few sec after having made the comment.
>
> In emails these updates get a mess,
> so better check my comments on github.
> Cheers
>
> Lukas
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/10763/files#r37070297>.
>



-- 
ANSHUL KAPOOR
Research Student at IISER Pune
-----Please Don`t Spam----------
anshul.kapoor@cern.ch
anshul.kapoor@students.iiserpune.ac.in